kv/memberlist/dnsprovider.go

```diff
@@ -0,0 +1,18 @@
+package memberlist
+
+import (
+	"context"
+)
+
+// DNSProvider supports storing or resolving a list of addresses.
+type DNSProvider interface {
+	// Resolve stores a list of provided addresses or their DNS records if requested.
+	// Addresses prefixed with `dns+` or `dnssrv+` will be resolved through respective DNS lookup (A/AAAA or SRV).
+	// For non-SRV records, it will return an error if a port is not supplied.
+	//
+	// Example: By providing the address of "dns+grafana.com:80", it should resolve into something like []string{"34.120.177.193:80", "[2600:1901:0:b3ea::]:80"}.
```

Comment:
@56quarters what do you think of this example? Couldn't come up with anything that produces results for `dnssrv+`.

Reply:
The prefixes documented here are implementation specific and shouldn't belong to the interface. The comment for this function should just be:

```
Resolve stores a list of provided addresses or their DNS records if requested.
```

Reply:
@pracucci thanks for clarifying that.

Reply:
I'll update it.